### ISSUE TITLE
Remove unnecessary restart variables

### ISSUE THREAD

**Issue by billsacks** [ESCOMP/ctsm#285]
**I originally made this comment 2015-10-29. I'm moving this over as is, though some of the original comment may no longer be correct.**

I have noticed that there are some variables on the restart file that are unnecessary. This adds confusion if you try to use the restart file as an indicator of the CLM state. This is particularly relevant for data assimilation.

Variables I have noticed are:

- [ ] plant_nalloc: seems to be set before it is used

- [ ] pot_f_nit_vr_vr and f_nit_vr_vr: I *think* they are set before they are used

- [ ] root_depth: I *think* it is set before it is used

- [ ] AnnET: Should be unnecessary because accumulators are automatically saved to restart

- [ ] qflx_floodg?: I haven't looked closely, but since this is set in lnd_import_export, it's hard to imagine why it's needed on the restart file. But more analysis or testing should be done before removing this.

- [ ] tws (see http://bugs.cgd.ucar.edu/show_bug.cgi?id=1731, now closed because I have moved this here)

- [ ] O2_DECOMP_DEPTH_SAT and O2_DECOMP_DEPTH_UNSAT?? (2016-05-19: I haven't looked closely, but at a glance it looks like they're not needed on restart; the same may be true of other variables in the methane model)

There are probably others, too

**Comment by billsacks** [comment]
@timhoar this one might be of interest to you

**Comment by slevisconsulting** [comment]
Getting myself set up according to prior templates. Listing the commands that I used in case something looks wrong to anyone:

cd /glade/work/slevis/git/clm5.0/
git clone https://github.com/ESCOMP/ctsm.git /glade/work/slevis/git/restart_var_cleanup
cd ../restart_var_cleanup/
./manage_externals/checkout_externals
git remote add escomp https://github.com/ESCOMP/ctsm.git   ! ...should have done "git rename"?
git remote add slevisconsulting https://github.com/ESCOMP/ctsm.git
git checkout -b restart_var_cleanup master

**Comment by slevisconsulting** [comment]
Thoughts about the method:

Running a case with as much of the model active as possible should give me access to the most complete variable list available.
Running a spun up case increases the likelihood that restart variables that should be updating, ARE updating.
Running a global simulation also increases the likelihood that restart variables are updating somewhere in the world during the test.

I will look for a standard test that fulfills these requirements or will come up with a new test.

Open to suggestions and comments.

**Comment by slevisconsulting** [comment]
Testing with no code mods first:
nohup qcmd -- ./create_test -f ../../cime_config/testdefs/ERP_intel_list.txt --project P93300641 --compare ctsm1.0.dev016 --walltime 01:00 &

**Comment by slevisconsulting** [comment]
@ekluzek , should I add the aforementioned file (ERP_intel_list.txt) to a commit when I open the PR, or should I leave it as untracked?

**Comment by billsacks** [comment]
> git remote add escomp https://github.com/ESCOMP/ctsm.git ! ...should have done "git rename"?

Yes, you can use git remote rename for this – or you can specify the origin name at clone time with the --origin flag to 'git clone'

> git remote add slevisconsulting https://github.com/ESCOMP/ctsm.git

You have a mistake here: this should point to your fork, not to escomp. You can fix it with git remote remove followed by a correct git remote add.

**Comment by billsacks** [comment]
@slevisconsulting I agree with your initial thoughts. There will be a trade-off here: If you run all ERP and ERS tests for every change, you're going to be waiting a while to see whether each change was successful, whereas if you just run one, then you risk having that test pass while another one would fail. I don't have a good answer for where you want to fall on that continuum. But my initial thought is: I'd probably identify a few short tests (multi-day global ERP or ERS tests probably covering at least Clm50BgcCrop, Clm50Sp, either Clm45Bgc or Clm45Cn, and maybe one or two others) that I could run if I want relatively fast feedback, but then, like you suggest, I'd run many or all of the ERP and ERS tests periodically, too – say, once I'd made a few changes and was done work for the day, I'd fire them off overnight... though I might just fire off the whole test suite at that point to be safe.

> should I add the aforementioned file (ERP_intel_list.txt) to a commit when I open the PR, or should I leave it as untracked?

You should leave it as untracked. If it just contains a subset of the full test list, there's no need to document it; if you end up running some additional tests that aren't included in the full test list, then you should document that in this issue or the PR.

And @slevisconsulting and @ekluzek - I'm happy to be the "point person" for reviewing changes for this issue.

**Comment by slevisconsulting** [comment]
> > git remote add escomp https://github.com/ESCOMP/ctsm.git ! ...should have done "git rename"?
> 
> Yes, you can use git remote rename for this – or you can specify the origin name at clone time with the --origin flag to 'git clone'
> 
> > git remote add slevisconsulting https://github.com/ESCOMP/ctsm.git
> 
> You have a mistake here: this should point to your fork, not to escomp. You can fix it with git remote remove followed by a correct git remote add.

I corrected with the following commands:

git remote remove escomp
git remote rename origin escomp
git remote remove slevisconsulting
git remote add slevisconsulting https://github.com/slevisconsulting/ctsm.git

If ok, then I will edit the list of commands in my earlier post to make it correct, in case I refer to it for help in the future.

**Comment by ekluzek** [comment]
@slevisconsulting and @billsacks just a note on an earlier issue. I agree with Bill that we shouldn't add "ERP_intel_list.txt" to master. But, if it's useful to you for the branch you could check it in temporarily on the branch, and then we remove it before coming to master. That's if there is a good reason to do that, you need the list on another machine, or you want us to look at it, etc.

On "git remote". Do "git remote -v" to check that your remotes are correct, and also do "git fetch --all" to make sure it can connect to all of them properly. If it can fetch from them you've got them setup right. It's easy enough to make a typo in a remote and it can't fetch.

@billsacks I'm happy if you want to be the point person. I suppose that means I'll just chime in when I have something to say, otherwise I'll wait for you to explicitly ask me.

**Comment by slevisconsulting** [comment]
Thanks, Erik, that helps!

**Comment by billsacks** [comment]
> I agree with Bill that we shouldn't add "ERP_intel_list.txt" to master. But, if it's useful to you for the branch you could check it in temporarily on the branch, and then we remove it before coming to master. That's if there is a good reason to do that, you need the list on another machine, or you want us to look at it, etc.

@ekluzek @slevisconsulting - my main reason for recommending against this is that I don't want people to get in the habit of adding temporary files to the repository that they revert later: In git, once you add a file, it's there in history forever, causing an increase in repository size for everyone cloning the repository moving forward, even if you remove the file in a later commit (unless you rewrite history via git rebase). For small-ish text files, this isn't a big deal, so as long as people can stay disciplined about only doing this for small-ish text files, it's fine. But, as a general rule, I'd like us to discourage people from doing this, because I think it's easier for people to remember, "Don't add temporary files" rather than "Don't add temporary files if they are large or binary files, but it's okay to add temporary small text files". It's a major pain to try to fix this problem after-the-fact, which is why I'd rather have people err on the side of being conservative here.

**Comment by ekluzek** [comment]
Good point @billsacks, thanks for giving the scoop on that.

**Comment by slevisconsulting** [comment]
Removed plant_nalloc from restart.
Running abbreviated test:

`qcmd -- ./create_test -f ../../cime_config/testdefs/ERP_intel_bgc_list.txt --project P93300641 --compare ctsm1.0.dev016 --walltime 06:00`

**Comment by slevisconsulting** [comment]
I created a similar directory named /restart_var_clnup2 with a branch by the same name.

I removed pot_f_nit_vr_vr  and f_nit_vr_vr from restart.
Running the same abbreviated test.

**Comment by slevisconsulting** [comment]
Absolutely.

**Comment by slevisconsulting** [comment]
Removing plant_nalloc PASSes the above test
while 
removing pot_f_nit_vr_vr and f_nit_vr_vr  FAILs
